pages/glossary/semantic-segmentation.md

```diff
@@ -0,0 +1,19 @@
+---
+title: Semantic segmentation
+key: semantic-segmentation
+unambiguous: true
+objective: false
+input_aspects:
+  - Accessibility tree
+  - CSS styling
+  - DOM tree
+---
+
+A [segmentation][] of an [HTML web page][] is a _semantic segmentation_ if there is no [block][] in the [segmentation][] that contain both nodes from inside and outside the [main block of content][] (if it exists).
+
+The [main block of content][] doesn't need to be a single [block] in the [segmentation][]. It may be split into sub-[blocks][block]. But it is not possible for a [block][] to "overlap" the boundaries of the [main block of content][].
```

Comment:
There is no requirement in the semantic segmentation to have one of its block being the "main block of content". Both the blocks of the segmentation and the main block of content are sets of nodes, but there is no reason why these sets should be the same.

The reason for this is that we could have a page with a huge main block of content (e.g., a full book) which is "naturally" split into several parts (e.g., each chapter). Then, the set of skiplinks at the start of the page could jump to each of the parts, which would actually be nice for users (instead of just jumping to the start of the main where you might be far from what you want to read). In that case, the semantic segmentation would have several blocks splitting the main block of content (one for each part), and the page could still pass "first focusable elements are links to sections of content".

I'm adding an example in that rule to illustrate this.

Reply:
I agree with everything you said. And I don't think any of it invalidates the point I was making :-)

According to the definition: 
> The _main block of content_ of an [HTML web page][] is a [block of content][]

This let us know 1) there is only one main block of content ("The main block of content" vs "A main block of content" if there were multiple) and 2) the main block is a block (not multiple blocks).

But in this definition you write:
> The [main block of content][] doesn't need to be a single [block]

About this sentence: 1) I don't think it is needed (what is the added value?) and 2) It contradicts the definition of main block by stating that it can be multiple blocks.

I still find the best would be to remove this paragraph.

Reply:
Blocks are just sets of nodes, and every set (well, almost) contains subsets. Similarly, almost every block contains sub-blocks.

If we take this example: https://github.com/act-rules/act-rules.github.io/pull/421/files#diff-6c9dd35ad79847fb9ce694ee339907c1R143-R180 there is only on "main block of content" (the `main` element), but it contains several sub-blocks. None of the sub-blocks are "main block of content", but they are nonetheless blocks.

The semantic segmentation that works for this example needs two sub-blocks inside the main block of content.

Of course, I have no problem removing this since it is indeed only clarification on something that is already a consequence of the previous paragraph 😀

Reply:
OTOH, and independently of this question, I am thinking "first focusable elements are links to sections" is maybe too strict.
I feel that blocks of content (repeated or not) that are after the main one do not need links to satisfy 2.4.1.

That is, taking https://github.com/act-rules/act-rules.github.io/pull/421/files#diff-6c9dd35ad79847fb9ce694ee339907c1R143-R180 again, if we remove the link to `bio-translator`, this would fail G124 but not 2.4.1 (I think). So Expectation 2 of this rule should only consider the main block of content and the blocks located before it. In which case, the possibility to split the main block of content in a semantic segmentation becomes pointless.

Given where semantic segmentation is used, it could become more strict by requiring one block to be the main block of content and one block to be everything after the main block of content (we don't care about this in 2.4.1). That might streamline a bit the rules using semantic segmentation. However, that might also make the definition a bit less reusable in the future…

WDYT?

Reply:
> Blocks are just sets of nodes, and every set (well, almost) contains subsets. Similarly, almost every block contains sub-blocks.
> [...]
> 
> The semantic segmentation that works for this example needs two sub-blocks inside the main block of content.

That is true so what about:

> The [main block of content][] might be [segmented][segmentation] into multiple sub-[blocks][block]. But it is not possible for such [blocks][block] to "overlap" the boundaries of the [main block of content][].

Reply:
> OTOH, and independently of this question, I am thinking "first focusable elements are links to sections" is maybe too strict.
> I feel that blocks of content (repeated or not) that are after the main one do not need links to satisfy 2.4.1.
> 
> That is, taking https://github.com/act-rules/act-rules.github.io/pull/421/files#diff-6c9dd35ad79847fb9ce694ee339907c1R143-R180 again, if we remove the link to `bio-translator`, this would fail G124 but not 2.4.1 (I think). So Expectation 2 of this rule should only consider the main block of content and the blocks located before it. In which case, the possibility to split the main block of content in a semantic segmentation becomes pointless.
> 
> Given where semantic segmentation is used, it could become more strict by requiring one block to be the main block of content and one block to be everything after the main block of content (we don't care about this in 2.4.1). That might streamline a bit the rules using semantic segmentation. However, that might also make the definition a bit less reusable in the future…
> 
> WDYT?

I agree that it is too strict. So I would be in favour of relaxing the rule. 
But I don't think we should change the definition. As you said, it might be easier to reuse it as it is.